tests/src/parallel_restore.c

```diff
@@ -596,14 +623,17 @@ static int do_restore(struct opts *opts)
 
 	ret = pread(dev_fd, super, SCOUTFS_BLOCK_SM_SIZE,
 		    SCOUTFS_SUPER_BLKNO << SCOUTFS_BLOCK_SM_SHIFT);
-	error_exit(ret != SCOUTFS_BLOCK_SM_SIZE, "error reading super, ret %d", ret);
+	error_exit(ret != SCOUTFS_BLOCK_SM_SIZE, "errorreading super, ret %d", ret);
 
 	ret = scoutfs_parallel_restore_create_writer(&wri);
 	error_exit(ret, "create writer %d", ret);
 
-	ret = scoutfs_parallel_restore_import_super(wri, super);
+	ret = scoutfs_parallel_restore_import_super(wri, super, dev_fd);
 	error_exit(ret, "import super %d", ret);
 
+	ret = scoutfs_parallel_restore_add_quota_rule(wri, generate_quotas(opts));
+	error_exit(ret, "add quotas %d", ret);
```

Comment:
Another thing to add is to implement quota restore in `restore_copy.c`.

Reply:
Cleaned up everything else, I'm going to push those changes up while I work on this